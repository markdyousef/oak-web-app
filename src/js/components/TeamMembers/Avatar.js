// @flow
import React, { PropTypes } from 'react';

import css from './Member';

// import placeholderImg
const placeholderImg = null;

const Avatar = (img: String) => (
    <div className={css.avatar}>
        <img src={img} alt="profile" />
    </div>
);

Avatar.propTypes = {
    img: PropTypes.string
};

Avatar.defaultProps = {
    img: placeholderImg
};

export default Avatar;
