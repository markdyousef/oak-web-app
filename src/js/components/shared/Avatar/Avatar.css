.avatar img {
    cursor: pointer;
    height: 100%;
    width: 100%;
    border-radius: 100px;
}
