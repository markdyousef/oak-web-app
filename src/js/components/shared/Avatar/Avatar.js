// @flow
import React, { PropTypes } from 'react';

import css from './Avatar.css';

const Avatar = ({ img }:Object) => (
    <div className={css.avatar}>
        {img && <img src={img} alt="profile" />}
    </div>
);

Avatar.propTypes = {
    img: PropTypes.string
};

Avatar.defaultProps = {
    img: null
};

export default Avatar;
