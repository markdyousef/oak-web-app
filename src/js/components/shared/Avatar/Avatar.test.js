// @flow
import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Avatar from './Avatar';

test('renders Avatar', () => {
    const props = {
        channels: [],
        getTeam: () => {},
        name: 'clai'
    };
    const shallowComponent = shallow(
        <Avatar {...props} />
    );
    expect(shallowComponent).toMatchSnapshot();
});
