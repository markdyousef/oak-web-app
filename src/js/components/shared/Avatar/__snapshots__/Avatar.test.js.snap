// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Avatar 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.img
    src={undefined}
    style={
        Object {
            "height": undefined,
            "width": undefined,
          }
    }
/>,
  "nodes": Array [
    <styled.img
      src={undefined}
      style={
            Object {
                  "height": undefined,
                  "width": undefined,
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        channels={Array []}
        getTeam={[Function]}
        name="clai"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "channels": Array [],
          "getTeam": [Function],
          "name": "clai",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.img
          src={undefined}
          style={
                    Object {
                              "height": undefined,
                              "width": undefined,
                            }
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <styled.img
          src={undefined}
          style={
                    Object {
                              "height": undefined,
                              "width": undefined,
                            }
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    channels={Array []}
    getTeam={[Function]}
    name="clai"
/>,
}
`;
